### ISSUE TITLE
config.json has to readable to unprivileged user 

### ISSUE THREAD

**Issue by kjetilk** [nodeSolidServer/node-solid-server#790]
During configuration to run solid-server under systemd, I couldn't find a way that `config.json` could be readable only by root when the system runs under an unprivileged user. Seemingly, the file has to be readable by the unprivileged user, which is bad.

**Comment by dmitrizagidulin** [comment]
@kjetilk can you tell me more about the expected permissions? Running solid-server as root is generally an anti-pattern (this is true of any service, though), so I'm not fully understanding this issue.

**Comment by kjetilk** [comment]
Right, I'll try to explain a bit further:

First, systemd is the system used in many modern Linux distros to manage the services that are spawned at boot time, as well as manage the lifecycle of services, to some extent I think also logging. It is quite nice.

The `systemd` dæmon itself is running as root, and so, when it starts a service like solid, it does so with root privileges. This makes it possible to have config files that are only readable to root, which is a good thing. However, as running the service itself as root is as you say a bad idea, root privileges should be dropped ASAP after launch. Normally, `systemd` also manages this really well, but with solid-server, I spent hours, getting errors I understood nothing of, trying "everything" and eventually gave up, and set 644 on `config.json`. `config.json` could potentially contain a clear text password to the SMTP server, so this is a bad thing. 

I don't quite remember the details anymore, but I hope this makes the issue clearer.

**Comment by melvincarvalho** [comment]
Just circling back on this one—it's quite an old issue, and from what I can tell, it relates more to deployment configuration than something specific to the node-solid-server codebase.

@kjetilk would it be OK to close this?

**Comment by kjetilk** [comment]
Yes, I suppose. I don't know if this is specific to NSS or a general problem with Node-based systems. If it is the latter, then that's pretty bad, if it is an NSS problem, it should be fixed, but then, I wouldn't run anything sensitive on NSS anyway, so I suppose it doesn't matter.
